crates/task-impls/src/quorum_proposal/dependency_handle.rs

```diff
@@ -175,6 +172,20 @@ impl<TYPES: NodeType> HandleDepOutput for ProposalDependencyHandle<TYPES> {
 
     #[allow(clippy::no_effect_underscore_binding)]
     async fn handle_dep_result(self, res: Self::Output) {
+        let high_qc_view_number = self.consensus.read().await.high_qc().view_number;
+        loop {
+            // Loop until the validated state map contains the view corresponding to the high QC.
+            if self
+                .consensus
+                .read()
+                .await
+                .validated_state_map()
+                .contains_key(&high_qc_view_number)
+            {
+                break;
+            }
+        }
```

Comment:
Does this loop make sense? In our original discussion, our conclusion was to wait for the `ValidatedStateUpdated` event, but I found it easier to block on the state map instead. Can we consider them equivalent?

Reply:
I think we can. For the purposes of what we're trying to accomplish, this seems more than adequate.

Reply:
I think this is going to cause really high CPU use as it's quite likely we are almost always going to be able to grab the read lock